### PR TITLE
Implement SLASensor

### LINKED ISSUE #166: [FEATURE] Implement an SLA Miss Sensor for workflows
**Is your feature request related to a problem? Please describe.**
Currently, brickflow has a health argument in the Workflow class that sets up an alert to trigger when a workflow exceeds a specified duration. This feature would add a sensor that allows users to specify a time by which the workflow is expected to be completed based on consumer service level agreements. 

**Cloud Information**

- [x] AWS
- [ ] Azure
- [ ] GCP
- [ ] Other

**Describe the solution you'd like**
A sensor task that monitors a given workflow's final task and sends an alert if that task has not completed by a user-specified time

**Describe alternatives you've considered**
We've considered just using the health rules, but those cannot be configured dynamically such that the duration is set for the amount of time between the workflow starting and the expected end time. (e.g. a workflow that's expected to complete within 2 hours will always have the same expected duration. if it needs to complete by 10am and starts at 9am, there's no way to dynamically adapt the duration down to one hour).

**Additional context**

### PR DESCRIPTION
**Pull request by chaelant**
## Description

This implements an SLASensor to monitor the on-time performance of Databricks workflows. It is a special instance of the WorkflowTaskDependencySensor that accepts an expected SLA time in the form of a UTC datetime object or a quartz cron expression in the workflow tags. The user provides the target workflow name and the task to monitor in that workflow that would indicate the workflow has completed. If the workflow does not finish by the specified time, an alert will be sent to a user-provided email and/or slack channel.

## Related Issue

This PR closes #166.

## Motivation and Context

Current alerting functionality only allows for alerts to be sent if a workflow runs longer than a specified amount of time. This change adds the functionality to be able to be alerted if a workflow does not finish by a certain time, regardless of when the workflow was kicked off. 

## How Has This Been Tested?

These changes were tested with `make test` and by building a wheel file and testing within a workflow. This implementation does not affect other features of brickflow. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: docs/faq/faq.md [hunk 1]
```diff
@@ -85,6 +85,93 @@ def wait_on_workflow(*args):
...
+        data_product="product_name",
+        run_date="2024-01-01",
+        sla_sensor_task_names=["sla_sensor"],
+        expected_sla_timestamp=datetime(2024, 1, 1, 10, 0, 0, tzinfo=pytz.utc),
+        dependency_job_name="target_job_name",
+        databricks_host="https://your_workspace_url.cloud.databricks.com",
+        databricks_token=api_token_key,
+        custom_description="message to provide additional context",
+        slack_webhook_url="https://hooks.slack.com/your/webhook/url",
+        email_params={
```
Comment (asingamaneni) [inline docs/faq/faq.md:114]: Can we make email_params in the class initialization as a pydantic object or a type object to validate the email_params.

Comment (chaelant) [reply docs/faq/faq.md:114]: Hi, @asingamaneni - updated `sla_sensor.py` to have an EmailParams pydantic object and use that to validate the email params passed to the SLASensor in the initialization. Will throw a `ValidationError` if missing required fields

File: brickflow_plugins/databricks/sla_sensor.py [hunk 2]
```diff
@@ -0,0 +1,596 @@
...
+import smtplib
+from brickflow import ctx
+from brickflow_plugins.databricks.workflow_dependency_sensor import (
+    WorkflowTaskDependencySensor,
+    WorkflowDependencySensorException,
+)
+
+
+class EmailParams(BaseModel):
+    email_list: str
```
Comment (asingamaneni) [inline brickflow_plugins/databricks/sla_sensor.py:20]: As we have the pydantic model now, probably it's easier for users to provide a list and we handle the rest.
Same goes for - cc as well.

If it is string and comma separated - after splitting, we also need to handle strip()

```suggestion
    email_list: List[str]
```

Let me know your thoughts, other wise the rest of the pr looks good to me

Comment (chaelant) [reply brickflow_plugins/databricks/sla_sensor.py:20]: That makes sense to me. Also is a little more intuitive from an end-user perspective. I'll make those updates

**Review by asingamaneni — APPROVED** [review]
LGTM

**Pull request merged**